test/unit/Site.test.js

```diff
@@ -0,0 +1,294 @@
+const Site = require('../../lib/Site');
+const path = require('path');
+const fs = require('fs-extra-promise');
+const ghpages = require('gh-pages');
+const cloneDeep = require('lodash/cloneDeep');
+
+const {
+  INDEX_MD_DEFAULT,
+  PAGE_EJS,
+  SITE_JSON_DEFAULT,
+  USER_VARIABLES_DEFAULT,
+} = require('./utils/data');
+
+jest.mock('fs');
+jest.mock('walk-sync');
+jest.mock('../../lib/Page');
+jest.mock('gh-pages');
+
+
+afterEach(() => fs.vol.reset());
+
+test('Site Generate builds the correct amount of assets', async () => {
+  const json = {
+    'lib/template/page.ejs': PAGE_EJS,
+    'inner/site.json': SITE_JSON_DEFAULT,
+
+    'asset/css/bootstrap.min.css': '',
+    'asset/css/github.min.css': '',
+    'asset/css/markbind.css': '',
+
+    'asset/js/setup.js': '',
+    'asset/js/vue.min.js': '',
+    'asset/js/vue-strap.min.js': '',
+
+    'asset/fonts/glyphicons-halflings-regular.eot': '',
+    'asset/fonts/glyphicons-halflings-regular.svg': '',
+    'asset/fonts/glyphicons-halflings-regular.ttf': '',
+    'asset/fonts/glyphicons-halflings-regular.woff': '',
+    'asset/fonts/glyphicons-halflings-regular.woff2': '',
+  };
+  fs.vol.fromJSON(json, '');
+  const site = new Site('inner/', 'inner/_site');
+  await site.generate();
+  const paths = Object.keys(fs.vol.toJSON());
+  const originalNumFiles = 13;
+  const expectedNumBuilt = 11;
+  expect(paths.length).toEqual(originalNumFiles + expectedNumBuilt);
+
+  // site
+  expect(fs.existsSync(path.resolve('inner/_site'))).toEqual(true);
```

Comment:
I think `.toEqual(true)` is same as `.toBeTruthy()`

ref: https://facebook.github.io/jest/docs/en/using-matchers.html#truthiness

Reply:
`.toEqual(true)` is a subset of `.toBeTruthy()`. I think the former is more correct.

Reply:
`toBeTruthy` tests if a value is true in a boolean context
`toBe` checks for exact equality using Object.is
`toEqual` is the same as toBe for Boolean values but for object it matches properties instead of identity 

```js
expect(1).toBeTruthy(); // passes
expect(1).toBe(true); // fails
expect(1).toEqual(true); // fails

const obj1 = { a: 1 };
const obj2 =  { a: 1 };

expect(obj1).toBe(obj2); // fails
expect(obj1).toEqual(obj2); // passes
```

Personally i would either use `.toBe` or `.toEqual`. Is it safe to assume that fs.existsSync will return a boolean? If that is the case then `.toBeTruthy` wlll be safe to use.